### DISCUSSION TITLE
Would Kubernetes Container Storage Interface (CSI) a valid use case for Cortx?

### DISCUSSION THREAD

**Discussion by kappa-wingman** [Seagate/cortx#372] [General] [answered]
Is Cortx suitable for Kubernetes (cloud native storage)?

Many object storage systems support K8S CSI https://kubernetes-csi.github.io/docs/drivers.html.
Would a CSI developed for use with K8S in the roadmap?

Thanks.

**Comment by mukul-seagate11** [comment]
@kappa-wingman, Yes would be a definitely a good use use for CORTX as we had already incorporated CSM component in Kubernetes but as a whole its still in roadmap

**Comment by kappa-wingman** [comment, marked as answer]
Great.

I guess a wider market share or adoption of open source storage products with CSI is rook-ceph and OpenEBS. Looking forward to see CORTX to support CSI in the future.

**Comment by johnbent** [comment]
Thank you @kappa-wingman !  We definitely want to support Kubernetes and CSI but we have some work to do to get there as we still have a small amount of in-kernel code running.  We have roadmap plans to replace those with user space and then Kubernetes and CSI are high-priority for us.  Thanks for feedback!
